src/canvastools/ts/CanvasTools/Core/RegionData.ts

```diff
@@ -145,6 +145,39 @@ export class RegionData implements IRegionData, IMovable, IResizable {
         return area;
     }
 
+    public getLineSegments(): Array<[Point2D, Point2D]> {
+        const points = this.regionPoints;
+        if (points.length < 2) {
+            return []
+        }
+        if (points.length === 2) {
+            return [[points[0], points[1]]]
+        }
+        const segments = [];
+        const pointsLength = points.length;
+        const loopLength = pointsLength - 1;
+        for (let i = 0; i < loopLength; i++) {
+            const nextPointIdx = i + 1;
+            if (nextPointIdx < pointsLength) {
+                segments.push([points[i], points[nextPointIdx]])
+            }
+        }
+        if (this.regionType == RegionDataType.Polygon) {
+            // closing line segment from last to first point
+            segments.push([points[pointsLength - 1], points[0]])
+        }
+        return segments;
+    }
+
+    public getLineMidpoints(): Point2D[] {
+        const lines = this.getLineSegments();
```

Comment:
I may be misusing GitHub's code review, but I only see this function (getLineSegments) used in one place.
1. Could we make it private?
2. We basically do two iterations here, first iteration to create the segments, second iteration to get midpoints. But it can be made into one iteration?